app/models/distribution.rb

```diff
@@ -54,6 +57,11 @@ def total_quantity
 
   private
 
+  def initialize_issued_at
+    self.issued_at ||= self.created_at
+    save
```

Comment:
You're right about the double-duty for the database -- perhaps instead of `after_create` it could be `before_create`? That would at least only fire this when the record is initially created and would do so before the database gets called up.